pkg/api/client.go

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+type CertificateResponse struct {
+	CertPEM  []byte
+	ChainPEM []byte
```

Comment:
worth parsing them into `x509.Certificate and []x509.Certificate` for increased usability?

Reply:
Lemme dig into how this is used on the `cosign` side more, since this came from there originally.  Generally SGTM, but I just want to avoid a nightmare picking up this change.

Reply:
Looks like these are passed around a fair amount.  First to `Signer` here: 

https://github.com/sigstore/cosign/blob/a47a835480c45d6db7a9751d7cb8bf87772799fc/cmd/cosign/cli/fulcio/fulcio.go#L185-L187

Next to `SignerVerifier` here:

https://github.com/sigstore/cosign/blob/7e5ff000f6af25793631eb590964b9a861d672dc/cmd/cosign/cli/sign/sign.go#L413-L415

Chasing the usages of the `SignerVerifier`'s `.Cert` it seems like the Rekor uploads want things like this, but only the policy code (to extract email/issuer) wants the structured form.


I'm mildly inclined to "Nack" on this, since it's a bit of plumbing and (given current usage) I'm not sure it helps usability a ton.  Maybe we can add an accessor to get the `x509.Certificate` later.  I can change if you feel strongly though, just LMK.